modules/EvManager/main/car_simulation.cpp

```diff
@@ -92,8 +94,59 @@ void CarSimulation::state_machine() {
         sim_data.state = SimState::UNPLUGGED;
         break;
     }
+
+    if (not state_has_changed and
+        (sim_data.state == SimState::CHARGING_REGULATED or sim_data.state == SimState::CHARGING_FIXED or
+         sim_data.state == SimState::ISO_CHARGING_REGULATED)) {
+        simulate_soc();
```

Comment:
Maybe this should already be triggered at startup, right now the ev_info is only published when a charging session is ongoing, but the SoC information is available all the time

Reply:
Yes, I think you could do this and only `sim_data.battery_charge_wh` will be increased during charging.